polkadot/runtime/common/src/paras_registrar/migration.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use frame_support::{
+	dispatch::GetStorageVersion,
+	traits::{Contains, OnRuntimeUpgrade, StorageVersion},
+};
+
+#[derive(Encode, Decode)]
+pub struct ParaInfoV1<Account, Balance> {
+	manager: Account,
+	deposit: Balance,
+	locked: bool,
+}
+
+pub struct MigrateToV1<T, UnlockParaIds>(sp_std::marker::PhantomData<(T, UnlockParaIds)>);
+impl<T: Config, UnlockParaIds: Contains<ParaId>> OnRuntimeUpgrade
+	for MigrateToV1<T, UnlockParaIds>
+{
+	fn on_runtime_upgrade() -> Weight {
+		let onchain_version = Pallet::<T>::on_chain_storage_version();
+
+		if onchain_version == 0 {
```

Comment:
We have a [`VersionedMigration`](https://github.com/paritytech/polkadot-sdk/blob/2f49252bcdb450c36f83a6dacdececc7b5d8e32c/substrate/frame/support/src/migrations.rs#L73) wrapper now that should do these checks.  
@liamaharon do you know if it also works when the pallet version is freshly initialized?

Reply:
Yes, it should work for new pallets

Reply:
It is still marked as `experimental`. Are we comfortable to use it with Polkadot?

Reply:
Yes, we've been using it for some time now for production polkadot and kusama migrations. I was actually going to suggest we remove the `experimental` flag because it makes it a bit cumbersome to use. @ggwpez think we're OK to remove it?